relay-common/src/constants.rs

```diff
@@ -204,7 +204,7 @@ impl From<EventType> for DataCategory {
 //
 // Note: This type is represented as a u8 in Snuba/Clickhouse, with Unknown being the default
 // value. We use repr(u8) to statically validate that the trace status has 255 variants at most.
-#[derive(Clone, Copy, Debug, PartialEq, Serialize)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize)]
```

Comment:
Let's revert this please. Most of our schema types do not implement `Eq` for a reason, because that allows us to add floats without breaking equality. There's also absolutely no disadvantage to not implementing `Eq` in practice. 

Ultimately, this comes down to API stability. I think we should disable this lint.

Reply:
yeah i wasn't entirely sure about that lint but we don't really have a stable api at all so i don't care too much.

Reply:
> There's also absolutely no disadvantage to not implementing Eq in practice.

I was surprised about this lint as well. Will remove it.